### ISSUE TITLE
Kernels Reject Input Textures > MAX_TEXTURE_SIZE/2

### ISSUE THREAD

**Issue by elgertam** [gpujs/gpu.js#552]
## *What* is wrong?
TL;DR: GPU.js allows input textures only up to the size [MAX_TEXTURE_SIZE/2, MAX_TEXTURE_SIZE/2].

WebGL(2)Kernel incorrectly validates input texture sizes before calculation. On my machine, the texture size limit is 16384. If I create a 2D array with 16384x16384 elements and try to pass it to a kernel, I get the following error: `gpu-browser.js:13154 Uncaught Error: Argument height of 32768 larger than maximum size of 16384 for your GPU`. 

## *Where* does it happen?
In the browser (Chrome/Firefox). Possibly also in node, but I haven't tested that yet.

## *How* do we replicate the issue?
0) Open console, import GPU.js, etc.
1) Create a matrix of size [MAX_TEXTURE_SIZE, MAX_TEXTURE_SIZE]
2) Create a kernel taking a matrix as input
3) Run the kernel with the matrix
4) See error in error console

I have created a jsFiddle that replicates these steps. https://jsfiddle.net/elgertam/y7fr61Lc/

## *How* important is this (1-5)?
4

## Expected behavior (i.e. solution)
I expect (and the error messages seem to indicate) that the kernel should run on a matrix/texture of size [MAX_TEXTURE_SIZE,MAX_TEXTURE_SIZE] or less.

**Comment by psishock** [comment]
This is working as intended I think.

The max texture size should be calculated as the sum of height*width.

Think about it, you may not work with perfectly square textures in some situations, but with custom rectangular ones. Nevertheless, the combination of any height*width shouldn't exceed the allowed MAX_TEXTURE_SIZE.

**Comment by elgertam** [comment]
@psishock That interpretation doesn't seem to square with my experience. From what I can tell, the width and height of a texture can each be at most MAX_TEXTURE_SIZE in WebGL(2). (Take a look e.g. at https://www.khronos.org/webgl/public-mailing-list/public_webgl/1307/msg00069.php and https://stackoverflow.com/questions/29975743/is-it-possible-to-use-webgl-max-texture-size.)

I created a new jsfiddle (modified from my original one) to further demonstrate why I think this is a bug https://jsfiddle.net/elgertam/ytcwjd2x/. The first execution should succeed with a texture having many more elements than MAX_TEXTURE_SIZE, while the second fails with a texture having slightly more than 1/4 elements of MAX_TEXTURE_SIZE^2. In neither case is the height or width of the texture close to MAX_TEXTURE_SIZE.

**Comment by psishock** [comment]
I loved the pun.

I'm not a webgl expert, but after reading your supplied links and verifying your jsfiddle, you seems to be exactly right.

**Comment by robertleeplummerjr** [comment]
Looking into this now.

**Comment by robertleeplummerjr** [comment]
Fixed in https://github.com/gpujs/gpu.js/releases/tag/2.4.8

**Comment by thesunfei** [comment]
> Fixed in https://github.com/gpujs/gpu.js/releases/tag/2.4.8

I still have this issue,it outputs an error:"Error: The required size exceeds that of the ability of your system",but the size is only half of  MAX_TEXTURE_SIZE.

**Comment by midnight-dev** [comment]
@thesunfei Can you post a public example from something like CodePen or JS Fiddle? Something that others can use to reproduce the issue.

**Comment by thesunfei** [comment]
> @thesunfei Can you post a public example from something like CodePen or JS Fiddle? Something that others can use to reproduce the issue.

https://codepen.io/thesunfei/pen/qBrNgWJ

You can see I set the canvas width to only half of maxTextureSize,it has render issue,but it works well if you change it to 1/4 of it.
